test/exchangeIssuance/exchangeIssuance-staged.spec.ts

```diff
@@ -0,0 +1,906 @@
+
+      const expectedUniRouterAddress = await exchangeIssuanceContract.uniRouter();
+      expect(expectedUniRouterAddress).to.eq(subjectUniswapRouter.address);
+
+      const expectedUniFactoryAddress = await exchangeIssuanceContract.uniFactory();
+      expect(expectedUniFactoryAddress).to.eq(subjectUniswapFactoryAddress);
+
+      const expectedSushiRouterAddress = await exchangeIssuanceContract.sushiRouter();
+      expect(expectedSushiRouterAddress).to.eq(subjectSushiswapRouter.address);
+
+      const expectedSushiFactoryAddress = await exchangeIssuanceContract.sushiFactory();
+      expect(expectedSushiFactoryAddress).to.eq(subjectSushiswapFactoryAddress);
+
+      const expectedControllerAddress = await exchangeIssuanceContract.setController();
+      expect(expectedControllerAddress).to.eq(subjectControllerAddress);
+
+      const expectedBasicIssuanceModuleAddress = await exchangeIssuanceContract.basicIssuanceModule();
+      expect(expectedBasicIssuanceModuleAddress).to.eq(subjectBasicIssuanceModuleAddress);
+    });
+
+    it("approves WETH to the uniswap and sushiswap router", async () => {
+      const exchangeIssuance: ExchangeIssuance = await subject();
+
+      // validate the allowance of WETH between uniswap, sushiswap, and the deployed exchange issuance contract
+      const uniswapWethAllowance = await setV2Setup.weth.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+      expect(uniswapWethAllowance).to.eq(MAX_UINT_256);
+
+      const sushiswapWethAllownace = await setV2Setup.weth.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+      expect(sushiswapWethAllownace).to.eq(MAX_UINT_256);
+
+    });
+  });
+
+  context("when exchange issuance is deployed", async () => {
+    let subjectWethAddress: Address;
+    let subjectUniswapFactoryAddress: Address;
+    let subjectUniswapRouter: UniswapV2Router02;
+    let subjectSushiswapFactoryAddress: Address;
+    let subjectSushiswapRouter: UniswapV2Router02;
+    let subjectControllerAddress: Address;
+    let subjectBasicIssuanceModuleAddress: Address;
+
+    let weth: WETH9;
+    let wbtc: StandardTokenMock;
+    let dai: StandardTokenMock;
+    let usdc: StandardTokenMock;
+
+    beforeEach(async () => {
+      let uniswapSetup: UniswapFixture;
+      let sushiswapSetup: UniswapFixture;
+
+      // TODO: should we instead port SystemFixtrue and use tokens from it ?
+      weth = setV2Setup.weth;
+      wbtc = setV2Setup.wbtc;
+      dai = setV2Setup.dai;
+      usdc = await deployer.setV2.deployTokenMock(user.address, 1000000 * 10 ** 6, 6, "USD Coin", "USDC");
+
+      uniswapSetup = await getUniswapFixture(owner.address);
+      await uniswapSetup.initialize(owner, weth.address, wbtc.address, dai.address);
+      sushiswapSetup = await getUniswapFixture(owner.address);
+      await sushiswapSetup.initialize(owner, weth.address, wbtc.address, dai.address);
+
+      subjectWethAddress = weth.address;
```

Comment:
I don't think these are necessary unless you plan on using them in the test suite below